spyder/widgets/sourcecode/codeeditor.py

```diff
@@ -1179,7 +1179,8 @@ def linenumberarea_paint_event(self, event):
         active_line_number = active_block.blockNumber() + 1
 
         def draw_pixmap(ytop, pixmap):
-            painter.drawPixmap(0, ytop + (font_height-pixmap.height()) / 2,
+            pixmap_height = pixmap.height() / pixmap.devicePixelRatio()

```

Comment:
`devicePixelRatio` was introduced in Qt5.5, that is making tests in Qt4 fail, maybe you could add a try.. except and leave the old behavior for Qt4